ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/snmp-settings.j2

```diff
@@ -143,6 +143,13 @@ snmp-server location {{ snmp_server.location }}
 {%     if snmp_server.traps.enable is arista.avd.defined(true) %}
 snmp-server enable traps
 {%     endif %}
+{%     for snmp_trap in snmp_server.traps.snmp_traps | arista.avd.natural_sort("name") %}
+{%         if snmp_trap.enabled is arista.avd.defined(false) %}
+no snmp-server enable traps {{ snmp_trap.name }}
+{%         else %}
```

Comment:
We should evaluate ``if snmp_trap.enabled is arista.avd.defined(true)``, otherwise config will be rendered with any value

Reply:
The default value of enabled is considered to be true. Mentioned it in the user guide. So if the user doesn't define enabled then also we have to render the config (default value is true). So I designed it this way. If I check true then I have initialize the enabled value to true and then check for user input if it's false. Anyway the user will give it as either true or false. Based on those values the configuration will be generated correctly.

Reply:
@xaviramon I see your point, and normally we don't have too many defaults in `eos_cli_config_gen`. I will ask @carlbuchmann for his opinion. @carlbuchmann check data model. We are discussing if `enabled` should be default `true` for specific `snmp_traps`

Reply:
In this case in the code we are just checking ``if snmp_trap.enabled is arista.avd.defined(false)`` which is ok. The problem is that we have an ``else`` statement, so if the value is false it will render ``no snmp-server enable traps {{ snmp_trap.name }}``, but if value is anything else ("yes","no","whatever") if will render the ``snmp-server enable traps {{ snmp_trap.name }}``, without being the variable "true"

Reply:
So we document to have the `default -> true` which makes it simple to use the data model. It will only take effect for someone actually entering the `name` key, so I think this is fine.